packages/demos/stories/TableDragDrop.tsx

```diff
@@ -0,0 +1,170 @@
+	 */
+	onMove: (dragIndex: number, hoverIndex: number) => void;
+}
+
+interface IDragItem {
+	/**
+	 * List Item unique identifier
+	 */
+	id: string;
+
+	/**
+	 * Index of the List Item in order to keep track of it's positioning within the array
+	 */
+	index: number;
+
+	/**
+	 * Type of data that is allowed to be dragged
+	 */
+	type: string;
+}
+
+const DraggableTableRow: React.FunctionComponent<IDraggableTableRowProps> = ({
+	id,
+	index,
+	onMove,
+}) => {
+	const ref = React.useRef<HTMLTableRowElement>(null);
+
+	const [, drop] = useDrop({
+		accept: 'tableRow',
+		hover(item: IDragItem, monitor: DropTargetMonitor) {
+			const dragIndex = item.index;
+
+			const hoverIndex = index;
+
+			if (!ref.current || dragIndex === hoverIndex) {
+				return;
+			}
+
+			const hoverBoundingRect = ref.current!.getBoundingClientRect();
+
+			const verticalMiddle =
+				(hoverBoundingRect.bottom - hoverBoundingRect.top) / 2;
+
+			const mousePosition = monitor.getClientOffset();
+
+			const pixelsToTop =
+				(mousePosition as XYCoord).y - hoverBoundingRect.top;
+
+			const draggingUpwards =
+				dragIndex > hoverIndex && pixelsToTop > verticalMiddle * 1.5;
+
+			const draggingDownwards =
+				dragIndex < hoverIndex && pixelsToTop < verticalMiddle / 2;
+
+			if (draggingDownwards || draggingUpwards) {
+				return;
+			}
+
+			onMove(dragIndex, hoverIndex);
+
+			item.index = hoverIndex;
+		},
+	});
+
+	const [{isDragging, itemBeingDragged}, drag] = useDrag({
+		collect: (monitor: any) => ({
+			isDragging: monitor.isDragging(),
+			itemBeingDragged: monitor.getItem() || {id: 0},
+		}),
+		item: {id, index, type: 'tableRow'},
+	});
+
+	const isItemBeingDragged = itemBeingDragged.id === id;
+
+	drag(drop(ref));
+
+	const style = {
+		backgroundColor: isItemBeingDragged ? '#EFEFEF' : '',
+		border: isItemBeingDragged ? '2px solid #555555' : '',
+		cursor: 'grab',
+		opacity: isDragging ? 0 : 1,
+	};
+
+	return (
+		<ClayTable.Row id={`tableRow${id}`} ref={ref} style={style}>
+			<ClayTable.Cell headingTitle>{id}</ClayTable.Cell>
+			<ClayTable.Cell>{'South America'}</ClayTable.Cell>
+			<ClayTable.Cell>{'Brazil'}</ClayTable.Cell>
+			<ClayTable.Cell truncate>
+				{
+					'Homero aeterno conclusionemque est in, scribentur mediocritatem mea ut. Ad voluptua vituperata constituam pro. Mel at constituto efficiantur, in eirmod lobortis mei. Eam vero probo efficiendi ne, molestie pericula nec in, dolore minimum duo et. Mundi epicuri patrioque in vis, virtute legimus oporteat cu eum.'
+				}
+			</ClayTable.Cell>
+		</ClayTable.Row>
+	);
+};
+
+const DragDroppableTable: React.FunctionComponent = () => {
+	const [tableRows, setTableRows] = React.useState([1, 2, 3]);
+
+	const onMove = React.useCallback(
+		(dragIndex: number, hoverIndex: number) => {
+			const tempList = [...tableRows];
+
+			tempList.splice(dragIndex, 1);
+
+			tempList.splice(hoverIndex, 0, tableRows[dragIndex]);
+
+			setTableRows(tempList);
+		},
+		[tableRows]
+	);
+
+	return (
+		<ClayTable>
```

Comment:
Same for this component, lets not use hardcoded values and instead pass in values like we might use in an app.